src/Symfony/Component/Config/Definition/Builder/TreeBuilder.php

```diff
@@ -41,6 +50,11 @@ public function root($name, $type = 'array', NodeBuilder $builder = null)
         return $this->root = $builder->node($name, $type)->setParent($this);
     }
 
+    public function getRootNode(): NodeDefinition
```

Comment:
this return type is broken when using the deprecated API and not calling `->root()`. I suggest throwing an exception with a meaningful message instead of letting PHP throw a TypeError (which will get reported as a bug in Symfony):

```php
throw new \RuntimeException(sprintf('Calling %s before creating the root node is not supported. Migrate to the new constructor signature instead', __METHOD__));
```